### DISCUSSION TITLE
Replace color

### DISCUSSION THREAD

**Discussion by jueqing1015** [dlemstra/Magick.NET#643] [Help]
Example: I need to replace all the red in a picture, including some similar ones, with white

The original ImageMagick has a similar operation, but in the MAGICK.NET Not found in
convert 2.jpg -alpha set -channel RGBA -fuzz 10% -fill "rgb(0,0,0)" -opaque "rgb(255,255,255)" new.png
Is it possible to integrate this function?

**Comment by dlemstra** [comment]
Your command would translate to this:

```C#
using (var image = new MagickImage(" 2.jpg:")) // 2.jpg
{
    image.Alpha(AlphaOption.Set); // -alpha set
    image.ColorFuzz = new Percentage(10); // -fuzz 10%
    image.Opaque(
        target: MagickColors.White, // -opaque "rgb(255,255,255)"
        fill: MagickColors.Black); // -fill "rgb(0,0,0)"
    image.Write("new.png"); // new.png
}
```

Not sure why you need the `-channel RGBA` option though.

**Reply by jueqing1015** [reply]
A problem about rotation

I rotate an image five degrees and I see that the image size gets bigger

Is there any way to rotate without changing the image size

If it is larger than the original size after rotation, remove the excess part directly

I don't know why the website can't upload images

**Reply by dlemstra** [reply]
You should be able to drag and drop or copy images into the messages here. And maybe @fmw42 knows a trick how we could rotate the image without the new background?

**Reply by jueqing1015** [reply]
In addition, I would like to ask if there is any way to increase the color of the text in a picture and change the color. The text is not necessarily black
And how to adjust the color scale

One of the following two images is that the text has turned red, the other is that the text has been accentuated as a whole. This is the effect that I use other software to achieve. I want to ask if our software can achieve this effect

Note: when the text becomes red, it is not necessarily black

![b](https://user-images.githubusercontent.com/65677359/82962572-2aabe880-9ff3-11ea-995f-0d81f0c37c54.jpg)
![a](https://user-images.githubusercontent.com/65677359/82962577-2da6d900-9ff3-11ea-9a4e-36e408594f82.jpg)

**Reply by fmw42** [reply]
This post is getting confusing. Is it about color change or about rotation. Please do not ask two questions in the same post. Make two separate posts in the future.

With regard to this question. I do not understand what you are asking. Do you need to change the color of the text or do you need to make text colors more saturated or do you need to make the text bolder or thicker? Please clarify your question.

**Reply by jueqing1015** [reply]
Sorry to pay attention next time

I want to change the color of the text and deepen the saturation of the color, but the key is that the color of the text on the image is not necessarily black

**Reply by fmw42** [reply]
Sorry, I am still not sure I understand. Do you want to deepen the saturation of the text color or the rest of the image color. Also please post and example input image, so we can try to demonstrate on that image.

**Comment by fmw42** [comment]
In command line mode, you can do that with -distort SRT

![lena](https://user-images.githubusercontent.com/13458218/82929441-7f3e5d80-9f38-11ea-9549-582ed8df4db9.jpg)

`convert lena.jpg -virtual-pixel white -distort SRT 10 lena_r10.jpg`

![lena_r10](https://user-images.githubusercontent.com/13458218/82929461-882f2f00-9f38-11ea-96fb-0267e141daf2.jpg)

See https://imagemagick.org/Usage/distorts/#srt
